server/src/com/cloud/network/router/CommandSetupHelper.java

```diff
@@ -703,6 +722,10 @@ public void createVpcAssociatePublicIPCommands(final VirtualRouter router, final
                     sourceNatIpAdd = new Pair<IpAddressTO, Long>(ip, ipAddr.getNetworkId());
                     addSourceNat = add;
                 }
+
+                if (!firstIP || add) {
+                    firstIP = false;
+                }
```

Comment:
for additional public subnet on delete it is not sure which ip is set to first ip. So on delete we want to set sourcenat to true for all ips to delete source nat rules. 

@ramkatru 
I will add above comment.